### PR TITLE
Fix/burn remote deadlock + burn fusion draining

### PR DESCRIPTION
**Pull request by nathanielsimard**
Improved a lot of things related to streams and training a model on multiple threads.

### CODE REVIEW

File: crates/burn-hip/src/lib.rs
Comment (laggui) [inline crates/burn-hip/src/lib.rs:32]: Not sure if this is meant to be commented out just locally or actually committed to fix some HIP stuff?

File: crates/burn-remote/src/client/worker.rs
Comment (laggui) [inline crates/burn-remote/src/client/worker.rs:108]: Debugging left overs or actual logs?

File: crates/burn-remote/src/client/worker.rs
Comment (laggui) [inline crates/burn-remote/src/client/worker.rs:129]: Debugging left overs or actual logs?

File: examples/text-classification/Cargo.toml
Comment (laggui) [inline examples/text-classification/Cargo.toml:45]: Left over? I don't see it being used in the example.

**Review by laggui — APPROVED** [review]
Approving in advance, just have some minor comments that might need to be addressed before merging.

File: crates/burn-hip/src/lib.rs
Comment (nathanielsimard) [reply crates/burn-hip/src/lib.rs:32]: It breaks rust-analyzer, so let's comment them until it's fixed.

**Pull request merged**
